packages/orchestrator-ui-components/src/components/WfoInlineNoteEdit/WfoSubscriptionDetailNoteEdit.tsx

```diff
@@ -0,0 +1,55 @@
+import type { FC } from 'react';
```

Comment:
This file is almost identical to `packages/orchestrator-ui-components/src/components/WfoInlineNoteEdit/WfoSubscriptionNoteEdit.tsx`. Is it possible to use generics?

Reply:
It will need a lot of if-checks do determine which query to use, also other type corrections.

Something like:
```
    const { data } = useGetSubscriptionDetailQuery({
        subscriptionId,
    },  {skip: !useQuery});

    const { selectedItem } = useQuery(queryVariables, {
        selectFromResult: (result: ApiResult<SubscriptionListResponse>) => ({
            selectedItem: result?.data?.subscriptions?.find(
                (sub) => sub.subscriptionId === subscriptionId,
            ),
        }),
    },
    );
```

```
    const updateMutation = !useQuery ? useUpdateSubscriptionDetailNoteOptimisticMutation : useUpdateSubscriptionNoteOptimisticMutation;

```